docs/source/bouncing.rst

```diff
@@ -0,0 +1,136 @@
+How PaaSTA Bounces New Code and SOA-config changes
+==================================================
+
+In the context of this document, "Bouncing" refers to the procedure of
+replacing old tasks with new ones. With long-running services bouncing
+is implemented by controlling Marathon Apps. For periodic tasks, see the
+section on "Chronos Bouncing."
+
+A "Bounce" can happen for one of these reasons:
+
+* A new version of the code is deployed (a git sha change)
+* A change of `soa-configs <yelpsoa_configs.html>`_ for a service. (Change in ram, cpu, environment variables, etc)
+* An issue of a ``paasta restart`` (or partially from a start/stop)
+* A change in system-wide PaaSTA configuration (defaults for volumes, ram, cpu, etc)
+
+By default, PaaSTA will do the safest thing possible and favors service uptime
+over speed. Read further for the actual details of this default procedure in the
+``crossover`` bounce.
+
+If service uptime is irrelevant then the ``brutal`` bounce can be selected.
+
+The ``upthendown`` and ``downthenup`` provide for other use cases where more
+predictable bouncing behavior is desired.
+
+Read more in the next section for exact details and differences amongst these
+bounce method.
+
+Note that in **no case** will PaaSTA "revert code or config" back to previous
+versions after a failed bounce. While some bounce methods have some protection
+against failure situations, PaaSTA will never undo the intent implied by a change.
+Addressing this problem should be done at at a layer *above PaaSTA*.
+
+
+Bounce Methods
+^^^^^^^^^^^^^^
+
+crossover
```

Comment:
Can we explicitly call out the fact that you can have up to 2x the desired number of instances running during a crossover bounce?